src/components/Tree/index.jsx

```diff
@@ -0,0 +1,193 @@
+    padding: 0,
+    display: 'block',
+  },
+  panelExpanded: {
+    margin: 0,
+    '&:before': {
+      opacity: 0,
+    },
+  },
+  childPanel: {
+    '&:before': {
+      opacity: 0,
+    },
+  },
+  text: {
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'noWrap',
+    maxWidth: '75vw',
+  },
+}))
+/**
+ * Render a tree view.
+ */
+export default class Tree extends Component {
+  static propTypes = {
+    /** The data to render as a tree view */
+    tree: arrayOf(shape(tree)).isRequired,
+    /** Callback function fired when a tree leaf is clicked. */
+    onLeafClick: func.isRequired,
+    /** A search term to refine the tree */
+    searchTerm: string,
+    /** Properties applied to the ExpansionPanelSummary element. */
+    expansionPanelSummaryProps: object,
+    /** Properties applied to the ExpansionPanelDetails element. */
+    expansionPanelDetailsProps: object,
+    /** Properties applied to the ListItem element. */
+    listItemProps: object,
+  };
+
+  static defaultProps = {
+    searchTerm: null,
+  };
+
+  createFilteredTree = memoizeWith(
+    (tree, searchTerm) => `${tree.toString()}-${searchTerm}`,
+    (tree, searchTerm) => (searchTerm ? this.filter(tree) : tree)
+  );
+
+  renderNode = (node, parent, depth = 0) => {
+    let treeDepth = depth;
+    const {
+      classes,
+      searchTerm,
+      onLeafClick,
+      expansionPanelSummaryProps,
+      expansionPanelDetailsProps,
+      listItemProps,
+      ...props
+    } = this.props;
+    const value = this.getNodeValue(node);
+    const isLeaf = this.isLeaf(node);
+    const { unit } = theme.spacing;
+    const textIndent = isLeaf
+      ? depth * unit + unit + (parent ? unit : 0)
+      : unit * depth + unit;
+
+    if (isLeaf && searchTerm && !value.includes(searchTerm)) {
+      return null;
+    }
+
+    if (isLeaf) {
+      return (
+        <ListItem
+          disableGutters
+          style={{ textIndent }}
+          key={value}
+          id={value}
+          value={value}
+          onClick={() => onLeafClick(value, parent)}
+          button
+          {...listItemProps}>
+          <div className={classes.text}>{value}</div>
+        </ListItem>
+      );
+    }
+
+    const expansionPanelClasses = {
+      expanded: classes.panelExpanded,
+      ...(parent ? { root: classes.childPanel } : null),
+    };
+
+    return (
+      <ExpansionPanel
+        className={classes.panel}
+        classes={expansionPanelClasses}
+        style={{ textIndent }}
+        key={node.value}
+        elevation={0}
+        {...props}>
+        <ExpansionPanelSummary
+          className={classes.panelSummary}
+          expandIcon={<ChevronDownIcon />}
+          {...expansionPanelSummaryProps}>
+          <div className={classes.text}>{node.value}</div>
+        </ExpansionPanelSummary>
+        <ExpansionPanelDetails
+          className={classes.panelDetails}
+          {...expansionPanelDetailsProps}>
+          {(treeDepth += 1) &&
```

Comment:
This mutates `treeDepth`. What does this do?

Reply:
No idea what I was doing here. Fixing this now.